xdem/volume.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+
+
+def hypsometric_binning(ddem: np.ndarray, dem: np.ndarray, bin_size=50,
+                        normalized_bin_size: bool = False) -> pd.DataFrame:
+    """
+    Separate the dDEM in discrete elevation bins.
+
+    :param ddem: The dDEM as a 2D or 1D array.
+    :param dem: The reference DEM as a 2D or 1D array.
+    :param bin_size: The bin interval size in georeferenced units (or percent; 0-100, if normalized_bin_size=True)
+    :param normalized_bin_size: If the given bin size should be parsed as a percentage of the glacier's elevation range.
+
+    :returns: A Pandas DataFrame with elevation bins and dDEM statistics.
+    """
+
+    assert ddem.shape == dem.shape
+    # Remove all nans, and flatten the inputs.
+    nan_mask = np.isnan(ddem) | np.isnan(dem)
```

Comment:
Note that now Raster's nodata are treated using masked arrays. So maybe it's worth mentioning in the docstrings that nodata values should be set to Nan prior to calling this function.
But maybe it's worth testing if the input array is of type np.masked_array and to use array.mask instead.